File: Emulator/src/main/core/net/packet/OutgoingPacket.java
package core.net.packet;

/**
 * Represents an outgoing packet.
 *
 * @author Emperor
 */
public interface OutgoingPacket<Context> {

    /**
     * Sends the packet.
     *
     * @param context The context.
     */
    public void send(Context context);

}

File: game-server/src/main/core/net/packet/OutgoingPacket.kt
package core.net.packet

/**
 * Represents an outgoing packet.
 * @author Emperor
 * @param Context The context type.
 */
interface OutgoingPacket<Context> {
    /**
     * Send.
     *
     * @param context the context
     */
    fun send(context: Context)
}